RFC: Extend `mountNode` prop in `Portal` (#27009)
microsoft/fluentui: rfcs/react-components/components/portal-mount-node.md

# RFC: Extend `mountNode` prop in `Portal`

[@layeshifter](https://github.com/layershifter)

## Summary

This RFC proposes extending the `mountNode` prop in the `Portal` component and its underlying components such as `Tooltip` and `Popup` to accept an object.

## Background

> The `Portal` component has a `mountNode` prop that allows customizing the element to which the `Portal` will be attached. However, there is no way to customize classes applied to that element. Customization is needed to apply styles such as custom `z-index`es. We need to be able to customize the `mountNode` element to apply this type of custom styles (([https://github.com/microsoft/fluentui/issues/26758](microsoft/fluentui#26758)).

Author: ```suggestion
The `Portal` component has a `mountNode` prop that allows customizing the element to which the `Portal` will be attached. However, there is no way to customize classes applied to that element. Customization is needed to apply styles such as custom `z-index`es. We need to be able to customize the `mountNode` element to apply this type of custom styles ([microsoft/fluentui#26758](https://github.com/microsoft/fluentui/issues/26758)).
```

> The `Portal` component has a `mountNode` prop that allows customizing the element to which the `Portal` will be attached. However, there is no way to customize classes applied to that element. Customization is needed to apply styles such as custom `z-index`es - `Portal` has a `mountNode` prop that allows to customize an element to which a `Portal` will be attached. We need to customize that element to apply custom styles, for example, custom `z-index`es (([https://github.com/microsoft/fluentui/issues/26758](microsoft/fluentui#26758)).

Reviewer: _nit:_ Remove repetitive statement

```suggestion
The `Portal` component has a `mountNode` prop that allows customizing the element to which the `Portal` will be attached. However, there is no way to customize classes applied to that element. Customization is needed to apply styles such as custom `z-index`es. We need to be able to customize the `mountNode` element to apply this type of custom styles (([https://github.com/microsoft/fluentui/issues/26758](microsoft/fluentui#26758)).
```
